scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/InProcessTaskExecutor.java

```diff
@@ -156,6 +160,14 @@ public TaskResultImpl execute(TaskContext taskContext, PrintStream output, Print
             if (nodesFile != null && !nodesFile.isEmpty()) {
                 FileUtils.deleteQuietly(new File(nodesFile));
             }
+            if (client != null) {
+                if (client.isConnected()) {
+                    try {
+                        client.disconnect();
+                    } catch (Exception ignored) {
```

Comment:
Not sure it is a good idea to ignore this exception. If disconnect fails, a session id may remains. To be able to have this exception in logs (at least trace or debug level) could help to understand the reason in case of debugging.